homeassistant/helpers/translation.py

```diff
@@ -321,3 +330,47 @@ async def async_get_translations(
     for entry in cached:
         result.update(entry)
     return result
+
+
+@bind_hass
+def get_cached_translations(
+    hass: HomeAssistant,
+    language: str,
+    category: str,
+    integrations: Iterable[str] | None = None,
+) -> dict[str, Any]:
+    """Return cached all backend translations.
+
+    If integrations specified, return translations for them.
+    Otherwise default to loaded integrations.
+    """
+    if integrations is not None:
+        components = set(integrations)
+    elif category in ("state", "entity_component", "services"):
+        components = set(hass.config.components)
+    else:
+        # Only 'state' supports merging, so remove platforms from selection
+        components = {
+            component for component in hass.config.components if "." not in component
+        }
+
+    result = {}
+
+    if TRANSLATION_FLATTEN_CACHE in hass.data:
+        cache = hass.data[TRANSLATION_FLATTEN_CACHE]
+        cached = cache.get_cached(language, category, components)
+        for entry in cached:
+            result.update(entry)
+
+    return result
```

Comment:
Replace with a guard returning early:
```suggestion
    if TRANSLATION_FLATTEN_CACHE not in hass.data:
        return {}
    
    result = {}

    cache = hass.data[TRANSLATION_FLATTEN_CACHE]
    cached = cache.get_cached(language, category, components)
    for entry in cached:
        result.update(entry)

    return result
```